ArtOfIllusion/src/artofillusion/script/ScriptWindowBase.java

```diff
@@ -0,0 +1,190 @@
+/* Copyright (C) 2020 by Maksim Khramov
+
+   This program is free software; you can redistribute it and/or modify it under the
+   terms of the GNU General Public License as published by the Free Software
+   Foundation; either version 2 of the License, or (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful, but WITHOUT ANY
+   WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+   PARTICULAR PURPOSE.  See the GNU General Public License for more details. */
+
+package artofillusion.script;
+
+import artofillusion.ui.Translate;
+import buoy.widget.AWTWidget;
+import buoy.widget.BComboBox;
+import buoy.widget.BFileChooser;
+import buoy.widget.BFrame;
+import buoy.widget.BMenu;
+import buoy.widget.BMenuBar;
+import buoy.widget.BorderContainer;
+import buoy.widget.LayoutInfo;
+import buoy.widget.RowContainer;
+import java.awt.Cursor;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.stream.Collectors;
+import javax.swing.JMenuItem;
+import javax.swing.JOptionPane;
+import javax.swing.filechooser.FileNameExtensionFilter;
+import org.fife.ui.rsyntaxtextarea.RSyntaxTextArea;
+import org.fife.ui.rsyntaxtextarea.SyntaxConstants;
+import org.fife.ui.rtextarea.RTextScrollPane;
+
+/**
+ *
+ * @author MaksK
+ */
+public class ScriptWindowBase extends BFrame
```

Comment:
Not a fan of this approach. As originally implemented, the script editor is simply a widget that provides the text editor functions. It composes well with any window arrangement that you want to embed it in.

As you've written this, one has to extend the `ScriptWindow`. Favor composition over inheritance.